calc/src/App.js

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from "react";
+import logo from "./logo.svg";
+// import "./App.css";
+import "./style.scss";
+class App extends Component {
+  state = {
+    calorie: 0,
+    done: false
+  };
+
+  calculate = e => {
+    e.preventDefault();
+    let calorieCalc;
+    let finalCalc = 0;
+    const height =
+      Number(e.target.inches.value) + Number(e.target.feet.value) * 12;
+
+    if (e.target.gender.value === "male") {
+      calorieCalc =
+        66 +
+        6.3 * e.target.weight.value +
+        12.9 * height -
+        6.8 * e.target.age.value;
+    } else {
+      calorieCalc =
+        655 +
+        4.3 * e.target.weight.value +
+        4.7 * height -
+        4.7 * e.target.age.value;
+    }
+
+    const activity = e.target.activity.value;
+    switch (activity) {
+      case "0":
+        finalCalc = calorieCalc * 1.12;
+        break;
+      case "1":
+        finalCalc = calorieCalc * 1.375;
+        break;
+      case "2":
+        finalCalc = calorieCalc * 1.55;
+        break;
+      case "3":
+        finalCalc = calorieCalc * 1.725;
+        break;
+      case "4":
+        finalCalc = calorieCalc * 1.9;
+        break;
+    }
+
+    this.setState({
+      done: true,
+      calorie: Math.floor(finalCalc)
+    });
+  };
+
+  render() {
+    return (
+      <main className="main">
+        <h1>Calorie Calculator</h1>
+        {this.state.done ? (
+          <section>
+            <h2>
+              Your recomended daily calorie intake for maintance is{" "}
+              {this.state.calorie}
+            </h2>
+            <p>
+              To lose a pound a week, it is recomended that you consume{" "}
+              {Math.floor(this.state.calorie - 71)} calories a day
+            </p>
+          </section>
+        ) : null}
+        <section>
+          <form onSubmit={this.calculate}>
```

Comment:
A way that you could refactor just a tiny bit is instead of that : null you could write 

{this.state.done && (html/JSX)} 

This will execute the html/JSX only if this.state.done is true and omits the unnecessary else null statement

Reply:
Thanks!